### ISSUE TITLE
Support negative coordinates, ie i16, i32, i64

### ISSUE THREAD

**Issue by Swoorup** [DoubleHyphen/morton-encoding#1]
One trick is to pre map this coordinates such that the negative minimum start at 0, 0.

**Comment by DoubleHyphen** [comment]
So you're saying to turn the coördinates to [offset-binary](https://en.wikipedia.org/wiki/Offset_binary) before encoding them, and then turn them back to one's complement after they're decoded? Hmmmmmmmm… this only entails flipping the most-significant bit, which in turn can be done in one CPU cycle using an XOR. Plus, if we do this after the key has been formed, we could XOR them all at once using a suitable bit-mask.

I guess either of us could do it. Do you have a specific use-case in mind, or are you speaking more generally?

**Comment by Swoorup** [comment]
Yeah just in general working with negative coordinates, nothing specific. I am currently using a [gist](https://gist.github.com/Swoorup/6fce2daf20385181e5dd4d674d9ae374) I created instead, due to other issue (performance). Using magic bytes instead of generating them by code appears to create an optimal assembly.

https://godbolt.org/z/deeaanjbW
